src/compiler/factory.ts

```diff
@@ -2587,18 +2587,28 @@ namespace ts {
         return node;
     }
 
-    export function createUnparsedSourceFile(text: string, map?: string): UnparsedSource {
+    export function createUnparsedSourceFile(text: string, mapPath?: string, map?: string): UnparsedSource {
         const node = <UnparsedSource>createNode(SyntaxKind.UnparsedSource);
         node.text = text;
+        node.sourceMapPath = mapPath;
         node.sourceMapText = map;
         return node;
     }
 
-    export function createInputFiles(javascript: string, declaration: string, javascriptMapText?: string, declarationMapText?: string): InputFiles {
+    export function createInputFiles(

```

Comment:
To avoid possible issues with existing calls, and since these aren't actually optional, we should have the types be `string | undefined` just to make sure we didn't miss any calls.